src/Shared/Cli.Utils/Command.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.Internal
+{
+    internal class Command
+    {
+        private readonly Process _process;
+        private bool _running = false;
+        private Action<string> _stdErrorHandler;
+        private Action<string> _stdOutHandler;
+
+        internal static Command CreateDotNet(string commandName, IEnumerable<string> args, NuGetFramework framework = null, string configuration = "Debug")
+        {
+            // Assuming that dotnet is on the PATH.
```

Comment:
Checkout: https://github.com/aspnet/DotNetTools/blob/feature/msbuild/shared/DotNetMuxer.cs. I want to move this into Microsoft.Extensions.CommandLineUtils soon.